File: in_Java/0383 Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] alpha = new int[26];
        for(char c : magazine.toCharArray()){
            alpha[c-'a']++;
        }
        for(char c : ransomNote.toCharArray()){
            alpha[c-'a']--;
        }
        for(int i : alpha){
            if(i < 0) return false;
        }
        return true;
    }
}

File: in_Python/0383 Ransom Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        dm = {}
        for m in magazine:
            if m not in dm:
                dm[m] = 1
            else:
                dm[m] += 1
        dr = {}
        for m in ransomNote:
            if m not in dr:
                dr[m] = 1
            else:
                dr[m] += 1
        for k in dr:
            if k not in dm:
                return False
            if dr[k] > dm[k]:
                return False
        return True